simpleGraph/bytesAlign.go

```diff
@@ -0,0 +1,422 @@
+				}
+				lastMatch = currMatch
+			} else {
+				//num++
+				//log.Printf("index=%d, vcfPos=%d\ndiff=%d\n%s", index, vcfsChr[i].Pos, vcfsChr[i].Pos-index, vcfsChr[i].Info)
+			}
+		}
+	}
+	//Case: last node
+	lastNode := &Node{Id: uint32(len(genome.Nodes)), Name: chr.Name, SeqTwoBit: dnaTwoBit.NewTwoBit(chr.Seq[index:]), Prev: make([]*Edge, 0, len(prev)), Next: nil, Info: &Annotation{Allele: 0, Start: uint32(index + 1), Variant: 0}}
+	AddNode(genome, lastNode)
+	for edge = 0; edge < len(lastMatch.Next); edge++ {
+		AddEdge(lastMatch.Next[edge].Dest, lastNode, 1)
+	}
+	if vcf.Snp(vcfsChr[len(vcfsChr)-2]) || isHaplotypeBlock(vcfsChr[len(vcfsChr)-2]) {
+		AddEdge(altAllele, lastNode, 1)
+	}
+	AddEdge(lastMatch, lastNode, 1)
+	SetEvenWeights(lastMatch)
+	return genome
+}
+
+func checkAlignment(aln *giraf.Giraf, genome *SimpleGraph) bool {
+	qName := strings.Split(aln.QName, "_")
+	//if len(qName) < 5 {
+	//	log.Fatalf("Error: input giraf file does not match simulation format...\n")
+	//}
+	if len(aln.Aln) < 1 {
+		return false
+	}
+
+	targetStart := aln.Path.TStart
+	targetEnd := aln.Path.TEnd
+	//if len(aln.Aln) < 1 {
+	if aln.ByteCigar[0].Op == 'S' {
+		//log.Printf("%s\n", giraf.GirafToString(aln))
+		targetStart = targetStart - int(aln.ByteCigar[0].RunLen)
+	}
+	if aln.Aln[len(aln.Aln)-1].Op == 'S' {
+		targetEnd = targetEnd + int(aln.ByteCigar[len(aln.ByteCigar)-1].RunLen)
+
+		//}
+
+	}
+	if common.StringToInt(qName[0]) == int(aln.Path.Nodes[0]) && common.StringToInt(qName[1]) == targetStart && targetEnd == common.StringToInt(qName[3]) {
+		//log.Printf("%s\n", giraf.GirafToString(aln))
+		//log.Printf("Results: %d != %d or %d != %d\n", headNode, aln.Path.Nodes[0], startPos, aln.Path.TStart)
+		//	log.Printf("%s\n", giraf.GirafToString(aln))
+		return true
+	} else {
+		//log.Printf("endPos=%d, right side cigar runLength: %d\n", endPos, aln.Aln[len(aln.Aln)-1].RunLen)
+		//log.Printf("%s\n", giraf.GirafToString(aln))
+		//log.Printf("Error: this read is not aligning correctly...\n")
+	}
+	return false
+}
+
+func isGirafPairCorrect(input <-chan *giraf.GirafPair, genome *SimpleGraph, wg *sync.WaitGroup, numReads int) {
+	var unmapped int = 0
+	for pair := range input {
+		if pair.Fwd.Aln == nil {
+			log.Printf("Warning... read was recorded as unmapped...\n")
+			continue
+		}
+		if !checkAlignment(pair.Fwd, genome) {
+			unmapped++
+			//log.Printf("Error: failed alignment simulation...\n")
+			//log.Printf("%s\n", giraf.GirafToString(pair.Fwd))
+			//log.Printf("%s\n", giraf.GirafToString(pair.Rev))
+		}
+		if !checkAlignment(pair.Rev, genome) {
+			//log.Printf("Error: failed alignment simulation...\n")
+			//log.Printf("%s\n", giraf.GirafToString(pair.Rev))
+			unmapped++
+		}
+	}
+
+	log.Printf("Mapped %d out of %d\n", numReads-unmapped, numReads)
+	log.Printf("%f of the reads are mapping correctly\n", percentOfFloat(numReads-unmapped, numReads))
+
+	wg.Done()
+}
+
+func ReadGenomeGraph(filename string) *SimpleGraph {
```

Comment:
I think this could even be in a separate file and pull request: 1) making the graph, 2) reading the graph, 3) aligning to the graph